compiler/src/compiler/cst_to_ast.rs

```diff
@@ -159,44 +163,133 @@ impl LoweringContext {
             }
             CstKind::Int { value, .. } => self.create_ast(cst.id, AstKind::Int(Int(value.clone()))),
             CstKind::Text {
-                opening_quote,
+                opening,
                 parts,
-                closing_quote,
+                closing,
             } => {
-                assert!(
-                    matches!(opening_quote.kind, CstKind::DoubleQuote),
-                    "Text needs to start with opening double quote, but started with {}.",
-                    opening_quote
-                );
+                let mut errors = vec![];
 
-                let text = parts
-                    .iter()
-                    .filter_map(|it| match it {
+                let opening_single_quote_count = match &opening.kind {
+                    CstKind::OpeningText {
+                        opening_single_quotes,
+                        opening_double_quote: box Cst {
+                            kind: CstKind::DoubleQuote,
+                            ..
+                        }
+                    } if opening_single_quotes
+                        .iter()
+                        .all(|single_quote| -> bool {
+                            matches!(single_quote.kind, CstKind::SingleQuote)
+                        }) => opening_single_quotes.len(),
+                    _ => panic!("Text needs to start with any number of single quotes followed by a double quote, but started with {}.", opening)
+                };
+
+                let mut lowered_parts = vec![];
+                for part in parts {
+                    match part {
                         Cst {
                             kind: CstKind::TextPart(text),
                             ..
-                        } => Some(text),
+                        } => {
+                            // TODO: Combine successive text parts during compile time. This can also be done in later stages.
```

Comment:
This gets done in the constant_folding stage of MIR optimization, so I think this TODO can get removed

Reply:
Resolved by 60922cc